compiler/optimizer/Structure.hpp

```diff
@@ -178,7 +178,8 @@ class TR_Structure
       if (f > SHRT_MAX-1)
          {
          TR_ASSERT(0, "nesting depth must be less than or equal to SHRT_MAX-1");
-         comp()->fe()->outOfMemory(comp(), "nesting depth must be less than or equal to SHRT_MAX-1");
+         traceMsg(comp(), "nesting depth must be less than or equal to SHRT_MAX-1");
+         throw TR::CompilationException();
```

Comment:
Should these be excessive complexity exceptions?  Or possibly their own exception type?